### PR TITLE
Fixed create message and show message bug.

### PR DESCRIPTION
**Pull request by the-spectator**
Renamed app/views/message to app/views/messages

### CODE REVIEW

File: app/controllers/messages_controller.rb [hunk 1]
```diff
@@ -9,11 +9,15 @@ def new
   def create
     m1 = message_params
     @email = m1[:receiver_id]
-    m1[:receiver_id] = User.find_by_email(@email).id
-    @message = current_user.sent_messages.new(m1)
-    @message.save!
-    if @message.save!
-      redirect_to dashboard_index_path
+    if User.exists?(email: @email)
```
Comment (anuj-verma) [inline app/controllers/messages_controller.rb]: We are making two queries here to database.
1. User.exists?
2. User.find

we can simply find user in 1 query and do the operation if it exists

File: app/controllers/messages_controller.rb [hunk 2]
```diff
@@ -9,11 +9,15 @@ def new
...
+      m1[:receiver_id] = User.find_by_email(@email).id
+      @message = current_user.sent_messages.new(m1)
+      @message.save!
```
Comment (anuj-verma) [inline app/controllers/messages_controller.rb]: why we are saving @message 2 times ?

File: app/controllers/messages_controller.rb [hunk 3]
```diff
@@ -9,11 +9,15 @@ def new
   def create
     m1 = message_params
     @email = m1[:receiver_id]
-    m1[:receiver_id] = User.find_by_email(@email).id
-    @message = current_user.sent_messages.new(m1)
-    @message.save!
-    if @message.save!
-      redirect_to dashboard_index_path
+    @receiver = User.find_by_email(@email)
+    if @receiver != nil
```
Comment (anuj-verma) [inline app/controllers/messages_controller.rb]: also we should not check `!= nil`

**Comment by Priyanka4328** [comment]
OK sir I will fix it

File: app/controllers/messages_controller.rb (hunk 3 above)
Comment (the-spectator) [reply app/controllers/messages_controller.rb]: For absent user User.find_by_email returns nil so was checking for that. What is more appropriate method, if i user user.exists? then there will be extra query.

**Pull request merged**